src/views/OpenedFilesView.tsx

```diff
@@ -0,0 +1,57 @@
+import { useDispatch } from 'react-redux';
+
+import { cx } from '../cx';
+import { closeFile, selectFile } from '../features/openedFiles/openedFilesSlice';
+import { useAppSelector } from '../redux/hooks';
+
+export function OpenedFilesView() {
+  const dispatch = useDispatch();
+  const openedFiles = useAppSelector((state) => state.openedFiles);
+
+  if (openedFiles.files.length === 0) return null;
+
+  return (
+    <div
+      className={cx(
+        'm-2 flex items-center gap-2', //
+        'border-b-1 border-b border-slate-400 ',
+      )}
+    >
+      {openedFiles.files.map((fileInfo) => (
+        <span
+          onClick={(evt) => {
+            evt.stopPropagation();
+            dispatch(selectFile(fileInfo));
+          }}
+          className={cx(
+            'flex flex-wrap items-center justify-between',
```

Comment:
With the caveat that I haven't used TailwindCSS before so it's hard for me to judge whether this is normal… the logic of the component really feels like it's getting drowned out by all these class names. Would it make sense to factor out a meaningfully-named component that applies most of these styles? https://tailwindcss.com/docs/reusing-styles#extracting-components-and-partials

Reply:
Sure. The focus of this PR was adding the state management (Redux) so I didn't added much ❤️ into that code. For the IDE issue I will fix this.